react-js/src/App.tsx

```diff
@@ -41,43 +41,92 @@ export default class App extends React.Component<any, any> {
+            overlay: {
+              visible: true,
+              onBarcodeFound: (code: Barcode, polygon: IBarcodePolygonHandle, label: IBarcodePolygonLabelHandle) => {
+                // You can override onBarcodeFound and create your own implementation for custom styling, e.g.
+                // if you wish to only color in certain types of barcodes, you can find and pick them, as demonstrated below:
+                if (code.format === "QR_CODE") {
+                  polygon.style({fill: "rgba(255, 255, 0, 0.3)", stroke: "yellow"})
+                }
+              }
+            },
+            // When dealing with AR overlay, let's hide the finder and have the whole area be detectable
+            showFinder: false
+          }}
+          onBarcodesDetected={this.onBarcodesDetected.bind(this)}
+        />
+      case RoutePath.MrzScanner:
+        return <MrzScannerComponent
+          onClose={onClose}
+          onMrzsDetected={this.onMrzDetected.bind(this)}
+        />;
+      case RoutePath.TextDataScanner:
+        return <TextDataScannerComponent
+          onClose={onClose}
+          onTextDataDetected={this.onTextDataDetected.bind(this)}
+        />;
+      case RoutePath.VINScanner:
+        return <VINScannerComponent
+          onClose={onClose}
+          onVINDetected={this.onVINDetected.bind(this)}
+        />;
+      case RoutePath.ScanAndCount:
+        return <BarcodeScannerComponent
+          onClose={onClose}
+          // To enable scan-and-count feature, add the additional config of scanAndCount: {}
+          additionalConfig={{scanAndCount: {enabled: true}, showFinder: false}}
+          hideCameraSwapButtons={true}
+          showBottomActionBar={false}
+          onBarcodesDetected={(barcodes: Barcode[]) => {
+            // Handle results as you please
+          }}
+        />;
+      default:
+        return null;
+    }
+  }
+
   async componentDidMount() {
     const sdk = await ScanbotSdkService.instance.initialize();
-    this.setState({ sdk: sdk });
+      this.setState({
+        sdk: sdk
+      });
 
     RoutingService.instance.observeChanges(() => {
       this.forceUpdate();
     });
 
     await ScanbotSdkService.instance.setLicenseFailureHandler((error: any) => {
-
       RoutingService.instance.reset();
-
-      this.setState({ error: { message: error } });
-      if (this._documentScanner?.isVisible()) {
-        this._documentScanner?.pop();
-      }
-      if (this._barcodeScanner?.isVisible()) {
-        this._barcodeScanner?.pop();
-      }
-      if (this._barcodeScannerWithOverlay?.isVisible()) {
-        this._barcodeScannerWithOverlay?.pop();
-      }
-      if (this._mrzScanner?.isVisible()) {
-        this._mrzScanner?.pop();
-      }
-      if (this._textDataScanner?.isVisible()) {
-        this._textDataScanner?.pop();
-      }
-      if (this._scanAndCounter?.isVisible()) {
-        this._scanAndCounter?.pop();
-      }
+      this.setState({error: {message: error}});
```

Comment:
formatting

Reply:
I think single-line objects are perfectly fine when the contents are simple, keeps the whole file cleaner. I just meant the first option, yeah, spaces are missing